### PR TITLE
Feature/update UI docs

### PR DESCRIPTION
**Pull request by debakkerb**
I updated the documentation based on feedback from participants.

### CODE REVIEW

**Review by guptamukul-google — COMMENTED** [review]
Can we move images under `radlab-ui/images` to a new folder under `docs/docs/rad-lab-ui/images` ? In that way all the documentation and related images will be at one place.

File: docs/docs/rad-lab-ui/overview.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+### Google Admin
+To grant access to users and to ensure that the application can validate group memberships, it's necessary to create groups and assign a custom role to service accounts when installing the RAD Lab UI.
+
+### Firebase 
+Invoking Firebase APIs via Infrastructure as Code from a terminal requires the use of a Google Cloud service account.  This is not supported by the RAD Lab UI, so the following steps have to be completed manually before the installation can be completed:
+- Create the Firebase application
+- Enable authentication mechanisms
+
+## Deploying a RAD Lab Module
```
Comment (guptamukul-google) [inline docs/docs/rad-lab-ui/overview.md:43]: This section should be moved under `docs/docs/rad-lab-ui/ui_deployment` where we talk about Module deploy / update & delete.

File: docs/docs/rad-lab-ui/ui_installation/source-control.md [hunk 2]
```diff
@@ -5,10 +5,16 @@ title: 01 - Source Control
 
 # Source Control
 
-We recommend forking the official [RAD Lab repository](https://github.com/GoogleCloudPlatform/rad-lab), to ensure that your organization keeps full control over the codebase.  For now, RAD Lab only supports GitHub, but this may change in the future.  When pushing the code to GitHub, ensure that it's configured to work together with the [Cloud Build integration](https://github.com/marketplace/google-cloud-build).  
+Users either have the option to fork the official [RAD Lab repository](https://github.com/GoogleCloudPlatform/rad-lab) or to clone the existing repository and push the code to a repository they manage and control.  The problem is that forking a repo results in a public repository, which may violate existing security policies in your organization.  
+
+For now, RAD Lab only supports GitHub, but this may change in the future.  When pushing the code to GitHub, ensure that it's configured to work together with the [Cloud Build integration](https://github.com/marketplace/google-cloud-build).
```
Comment (guptamukul-google) [inline docs/docs/rad-lab-ui/ui_installation/source-control.md:10]: Is Cloud Build and GitHub integration mandatory or just an optional step ?

File: docs/docs/rad-lab-ui/ui_installation/infrastructure.md [hunk 3]
```diff
@@ -127,35 +122,7 @@ The reason why you need to run `terraform init` twice is that the Terraform code
 a `terraform {}` resource, that points to a newly created bucket in the RAD Lab UI project. This will allow you to store
 the state remotely and collaborate on any future changes with other developers.
 
-It's important to point out that `terraform.tfvars` contains sensitive information. It's up to you to decide whether or
-not you want to store this in Github. If that is not the case, simply don't add that file.
-
-:::danger Important
```
Comment (guptamukul-google) [inline docs/docs/rad-lab-ui/ui_installation/infrastructure.md:133]: Any reason to remove this section ?

File: docs/docs/rad-lab-ui/ui_installation/source-control.md [hunk 4]
```diff
@@ -5,10 +5,16 @@ title: 01 - Source Control
 
 # Source Control
 
-We recommend forking the official [RAD Lab repository](https://github.com/GoogleCloudPlatform/rad-lab), to ensure that your organization keeps full control over the codebase.  For now, RAD Lab only supports GitHub, but this may change in the future.  When pushing the code to GitHub, ensure that it's configured to work together with the [Cloud Build integration](https://github.com/marketplace/google-cloud-build).  
+Users either have the option to fork the official [RAD Lab repository](https://github.com/GoogleCloudPlatform/rad-lab) or to clone the existing repository and push the code to a repository they manage and control.  The problem is that forking a repo results in a public repository, which may violate existing security policies in your organization.
```
Comment (guptamukul-google) [inline docs/docs/rad-lab-ui/ui_installation/source-control.md:8]: Can we state that we recommend to **clone the repo** and **move it to their own Private repo**?

**Review by guptamukul-google — COMMENTED** [review]
Can we also add the section on how one can get the bug fixes and new feature updates from Public RAD Lab repo to their own private repos.

File: docs/docs/rad-lab-ui/ui_installation/infrastructure.md [hunk 5]
```diff
@@ -18,14 +18,9 @@ proceeding
 
 The following prerequisites have to be met in order to proceed with the Terraform installation.
 
-### Resource Hierarchy
```
Comment (guptamukul-google) [inline docs/docs/rad-lab-ui/ui_installation/infrastructure.md:21]: I think we can keep this section. Any reason to remove it ?

File: docs/docs/rad-lab-ui/ui_installation/infrastructure.md [hunk 6]
```diff
@@ -176,19 +143,13 @@ Personal Access Token) to fetch the RAD Lab modules from your repo.
...
 
-You can add a secret directly on the command line using below command (replace `${GIT_PERSONAL_ACCESS_TOKEN}` with the token copied from the Github UI) :
+You can add a secret directly on the command line using below command (replace `${GIT_PERSONAL_ACCESS_TOKEN}` with the token copied from the Github UI), where `${PROJECT_ID}` corresponds to the RAD Lab UI project ID. 
 
 ```bash
 echo -n "${GIT_PERSONAL_ACCESS_TOKEN}" | \
     gcloud secrets versions add $(terraform output -json | jq -r .git_personal_access_token_secret_id.value) --data-file=- --project ${PROJECT_ID}
 ```
 
-To get `PROJECT_ID`, run below command. Copy output and replace `${PROJECT_ID}`
```
Comment (guptamukul-google) [inline docs/docs/rad-lab-ui/ui_installation/infrastructure.md:186]: We can still keep this section as it directly gives the Project ID

**Review by guptamukul-google — CHANGES_REQUESTED** [review]
Left few comments. Images looks great for users. Thank you!

**Comment by debakkerb** [comment]
> Can we move images under `radlab-ui/images` to an existing new folder - `docs/static/img` ? In that way all the documentation and related images will be at one place.

I don't mind, but I'd argue that that folder pertains to documentation itself, while these images are about the UI.  I replicated the implementation from the modules, as they all have an `images`-folder, with `architecture.png` inside.  Let me know if you still want me to move them or not.

File: docs/docs/rad-lab-ui/ui_installation/source-control.md (hunk 2 above)
Comment (debakkerb) [reply docs/docs/rad-lab-ui/ui_installation/source-control.md:10]: It's mandatory, the UI won't work without it

File: docs/docs/rad-lab-ui/ui_installation/infrastructure.md (hunk 6 above)
Comment (debakkerb) [reply docs/docs/rad-lab-ui/ui_installation/infrastructure.md:186]: After running the enablement sessions, it's incredibly confusing for people who go through the instructions, as they think they have to re-run that statement.  That's why I removed it.

File: docs/docs/rad-lab-ui/overview.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+- Google App Engine application to host the UI.
+- Firebase application to manage routing, security, etc.
+- Google Cloud Storage buckets for the Terraform state files, for both the UI and the individual RAD Lab modules.
+- Necessary service accounts with IAM permissions, needed for their activities.
+- Cloud Build triggers to deploy the RAD Lab modules.
+
+### Shell scripts
+The Cloud Admin will have to run a number of Shell scripts to configure the Firebase application, incl. Firestore rules and configuration specific to the UI.
+
+### Manual
```
Comment (acolver) [inline docs/docs/rad-lab-ui/overview.md:29]: ### Manual Tasks

File: docs/docs/rad-lab-ui/overview.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+- Google Cloud project, including APIs.
+- A number of Cloud Functions, which are used by the RAD Lab UI to create RAD Lab Module.
+- Google App Engine application to host the UI.
+- Firebase application to manage routing, security, etc.
+- Google Cloud Storage buckets for the Terraform state files, for both the UI and the individual RAD Lab modules.
+- Necessary service accounts with IAM permissions, needed for their activities.
+- Cloud Build triggers to deploy the RAD Lab modules.
+
+### Shell scripts
+The Cloud Admin will have to run a number of Shell scripts to configure the Firebase application, incl. Firestore rules and configuration specific to the UI.
```
Comment (acolver) [inline docs/docs/rad-lab-ui/overview.md:27]: I would prefer no `.` at the end of these (as most aren't sentences).

File: docs/docs/rad-lab-ui/overview.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+
+### Steps
+
+When users create a RAD Lab module, the following steps are being executed by the underlying backend components:
+1. Users select a module from the UI and click Create, supplying the necessary values for the Terraform variables.
+2. The UI posts a message to a Pub/Sub topic, passing the values for the variables in `variables.tf` for that specific module and some [additional information](../../../radlab-ui/automation/terraform/infrastructure/function/create_deployment/index.js).
+3. The Cloud Function creates 2 files, `backend.tf` and `terraform.tfvars.json`.  `backend.tf` contains information regarding Terraform remote state for that specific module creation and `terraform.tfvars.json` contains the values for `variables.tf` for that specific module.
+4. Once the previous step is completed, the Cloud Function invokes the Cloud Build trigger to create the RAD Lab modules and stores the unique build ID from Cloud Build in Firestore.
+5. The RAD Lab UI retrieves the build ID for that specific RAD Lab module and polls the Cloud Build APIs for the build status and logs for that particular RAD Lab module and updates the UI for the end user.
```
Comment (acolver) [inline docs/docs/rad-lab-ui/overview.md:56]: Best to have ordered lists all start with `1.`, that way you can reorder without having to update the order manually.

File: docs/docs/rad-lab-ui/overview.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+![](../../../radlab-ui/images/flow.png)
+
+The diagram depicts the flow to deploy a RAD Lab module, as a RAD Lab user, via the UI.  The steps to update and delete a module follow a similar pattern, so they are not repeated here. 
+
+### Steps
+
+When users create a RAD Lab module, the following steps are being executed by the underlying backend components:
+1. Users select a module from the UI and click Create, supplying the necessary values for the Terraform variables.
+2. The UI posts a message to a Pub/Sub topic, passing the values for the variables in `variables.tf` for that specific module and some [additional information](../../../radlab-ui/automation/terraform/infrastructure/function/create_deployment/index.js).
+3. The Cloud Function creates 2 files, `backend.tf` and `terraform.tfvars.json`.  `backend.tf` contains information regarding Terraform remote state for that specific module creation and `terraform.tfvars.json` contains the values for `variables.tf` for that specific module.
```
Comment (acolver) [inline docs/docs/rad-lab-ui/overview.md:54]: 1. The Cloud Function creates two files, `backend.tf` and `terraform.tfvars.json`.  `backend.tf` contains information regarding Terraform remote state for that specific module creation, and `terraform.tfvars.json` contains the values for `variables.tf` for that specific module.

File: docs/docs/rad-lab-ui/overview.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+4. Once the previous step is completed, the Cloud Function invokes the Cloud Build trigger to create the RAD Lab modules and stores the unique build ID from Cloud Build in Firestore.
```
Comment (acolver) [inline docs/docs/rad-lab-ui/overview.md:55]: 1. Once the previous step is completed, the Cloud Function invokes the Cloud Build trigger to create the RAD Lab module and stores the unique build ID from Cloud Build in Firestore.

File: docs/docs/rad-lab-ui/overview.md (hunk 9 above)
Comment (acolver) [inline docs/docs/rad-lab-ui/overview.md:56]: 1. The RAD Lab UI retrieves the build ID for that specific RAD Lab module and polls the Cloud Build APIs to display the build status and logs.

File: docs/docs/rad-lab-ui/ui_installation/frontend.mdx [hunk 12]
```diff
@@ -60,19 +58,50 @@ Add your Project ID here to have the documentation be be customized to your Goog
 
 <ProjectIdInput />
 
-## Add Firebase
+# Add Firebase
+
+:::warning
+If you've never used Firebase in your Google Cloud organization before, please execute [First Time Use](#first-time-use).  If you already have Firebase projects running in your Google Cloud organization or you have in the past, please execute [Returning Firebase User](#returning-firebase-user).
```
Comment (acolver) [inline docs/docs/rad-lab-ui/ui_installation/frontend.mdx:64]: If you've never used Firebase in your Google Cloud organization before, please follow [First Time Use](#first-time-use).  If you already have Firebase projects running in your Google Cloud organization or you have in the past, please follow [Returning Firebase User](#returning-firebase-user).

File: docs/docs/rad-lab-ui/ui_installation/frontend.mdx [hunk 13]
```diff
@@ -60,19 +58,50 @@ Add your Project ID here to have the documentation be be customized to your Goog
...
 
-:::important First time
-If you have never used Firebase in your Google Cloud organization, you need to manually add it to the Google Cloud project created in the first step. This is necessary as you have to accept the Terms & Conditions first.
+If you're not sure, simply execute the steps as described in [First Time Use](#first-time-use).
```
Comment (acolver) [inline docs/docs/rad-lab-ui/ui_installation/frontend.mdx:66]: If you're not sure, simply follow the steps as described in [First Time Use](#first-time-use).

File: docs/docs/rad-lab-ui/ui_installation/frontend.mdx [hunk 14]
```diff
@@ -60,19 +58,50 @@ Add your Project ID here to have the documentation be be customized to your Goog
...
+
+```shell
+# Replace [PROJECT_ID] with the actual project ID.
+export RLUI_PROJECT_ID=$(terraform show -json | jq -r .values.outputs.project_id.value)
+
+# Install Firebase Tools
+npm install -g firebase-tools
+
+# Switch to the webapp directory
+cd ../../../webapp
```
Comment (acolver) [inline docs/docs/rad-lab-ui/ui_installation/frontend.mdx:85]: It's unclear to me which working directory I should be in when I run `cd ../../../webapp`

File: docs/docs/rad-lab-ui/ui_installation/frontend.mdx [hunk 15]
```diff
@@ -60,19 +58,50 @@ Add your Project ID here to have the documentation be be customized to your Goog
...
+firebase deploy --only firestore:rules
+
+# Upload Firestore indexes
+firebase deploy --only firestore:indexes
+```
+
+## Returning Firebase User
+
+:::warning first time
+Only execute these steps if you already have one or more Firebase applications in your Google Cloud organization.  If that is not the case, or you are not sure, please run the steps in the previous section, titled **First time Firebase use**.
```
Comment (acolver) [inline docs/docs/rad-lab-ui/ui_installation/frontend.mdx:103]: Please provide a link

**Review by acolver — CHANGES_REQUESTED** [review]

File: docs/docs/rad-lab-ui/ui_installation/frontend.mdx (hunk 14 above)
Comment (debakkerb) [reply docs/docs/rad-lab-ui/ui_installation/frontend.mdx:85]: Normally users can just follow along with these steps (copy/paste).  I'll specify in that comment what directory they should move to, but out of experience I know users don't read that comment :).

**Comment by debakkerb** [comment]
> Can we also add the section on how one can get the bug fixes and new feature updates from Public RAD Lab repo to their own private repos.

I've never done this, so if you can give me a bullet list of steps to execute, I can add them.  Or feel free to add them yourself :)

**Review by guptamukul-google — APPROVED** [review]
Changes looks good to me!

**Pull request merged**